packages/es-components/src/components/containers/heading/Heading.tsx

```diff
@@ -1,23 +1,48 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import styled, { css } from 'styled-components';
+import styled, {
+  DefaultTheme,
+  ThemedStyledProps,
+  css
+} from 'styled-components';
+import { HeadingLevel } from '../../../../../../shared/types/src/ESTheme';
 
 const knockoutStyles = css`
   background-color: ${({ theme }) => theme.colors.primary};
   color: white;
   padding: 20px 15px;
 `;
 
-const getAdjustedProps = func => ({ size, level = 1, ...props }) =>
-  func({
-    ...props,
+export type HeadingProps = Omit<JSX.IntrinsicElements['h1'], 'ref'> & {
```

Comment:
There is a global helper we added for this.
```suggestion
export type HeadingProps = JSXElementProps<'h1'> & {
```